patient-appointment-management/functions/login.js

```diff
@@ -2,42 +2,44 @@ exports.handler = function (context, event, callback) {
   const { path } = Runtime.getFunctions().auth;
   const { isValidPassword, createMfaToken } = require(path);
 
-  const response = new Twilio.Response();
-  response.appendHeader('Content-Type', 'application/json');
+  async function sendMfaCode(context) {
+    const path0 = Runtime.getFunctions().helpers.path;
+    const { getParam } = require(path0);
+    context.TWILIO_VERIFY_SID = await getParam(context, 'TWILIO_VERIFY_SID');
 
-  if (isValidPassword(event.password, context)) {
-    // Generate and send six digit MFA code to administrator's phone
     const twilioClient = context.getTwilioClient();
-    mfaCode = Math.floor(100000 + Math.random() * 900000);
-    twilioClient.messages
-      .create({
+    const channel = 'sms';
+    return twilioClient.verify
+      .services(context.TWILIO_VERIFY_SID)
+      .verifications.create({
         to: context.ADMINISTRATOR_PHONE,
-        from: context.TWILIO_PHONE_NUMBER,
-        body: `${mfaCode} is your security code for the PAM application. It is valid for five minutes.`,
-      })
-      .then(function () {
-        // SMS was sent successfully
-        response.setBody({
-          token: createMfaToken('login', mfaCode, context),
-        });
-
-        callback(null, response);
-      })
-      .catch(function () {
-        // if there is any problem in sending SMS
-        response.setStatusCode(500);
-        response.setBody({
-          message:
-            'Unable to send security code to your phone at this time. Please try later.',
-        });
-        callback(null, response);
+        channel,
       });
-    return;
   }
 
-  // eslint-disable-next-line no-undef
-  response.setStatusCode(401);
-  response.setBody({ message: 'Unauthorized' });
+  const response = new Twilio.Response();
+  response.appendHeader('Content-Type', 'application/json');
 
-  callback(null, response);
+  if (!isValidPassword(event.password, context)) {
+    response.setStatusCode(401);
+    response.setBody({ message: 'Unauthorized' });
+    return callback(null, response);
+  }
+
+  sendMfaCode(context)
+    .then(function () {
+      response.setBody({
+        token: createMfaToken('login', context),
```

Comment:
What's the benefit of creating the token at this point? I'm not clear on how it's being used. I see there's also `createAppToken` once the mfa code is verified, why do we need a different token for the interim?

Reply:
We are using Twilio verify as a second factor authentication (the first factor is the user's password). The MFA token is issued when the user is password-authenticated but not  yet twilio-verified. This token will only allow MFA validation and not  give access to  App functionality. An **App token** is issued when both steps are complete. This new token gives access to all app functionality. Let us know if we can clarify this any further.

Reply:
makes sense, thanks!